energonai/model/model_generation_factory.py

```diff
@@ -0,0 +1,310 @@
+import os
+import time
+import torch
+import random
+from torch import nn, dtype
+from typing import Callable, Optional
+from .endecoder import Block1D
+from .embedding import Embedding1D
+from .downstream import LMHead1D
+
+from energonai.nn import LayerNorm1D
+from colossalai.core import global_context as gpc
+from colossalai.context import ParallelMode
+from energonai.utils import is_using_pp, get_current_device
+from energonai.logging import get_dist_logger
+from energonai.utils.checkpointing import load_checkpoint
+from energonai.utils.checkpointing_hf_gpt2 import processing_HF_GPT
+from energonai.utils.checkpointing_opt import processing_OPT
+from transformers.generation_logits_process import TopKLogitsWarper, TopPLogitsWarper, TemperatureLogitsWarper, LogitsProcessorList
+
+
+def gelu_impl(x):
+    """OpenAI's gelu implementation."""
+    return 0.5 * x * (1.0 + torch.tanh(0.7978845608028654 * x *
+                                       (1.0 + 0.044715 * x * x)))
+
+
+def select_top_k(predictions, k=5):
+    predicted_index = random.choice(predictions[0, -1, :].sort(descending=True)[1][:10])  # .item()
+    return predicted_index
+
+
+class PipelineModel(nn.Module):
```

Comment:
Could you inherit the class in model factory, instead of copy & paste?

Reply:
I have done so.